pkg/agent/controller/egress/egress_controller.go

```diff
@@ -1169,9 +1169,14 @@ func (c *EgressController) uninstallEgress(egressName string, eState *egressStat
 		}
 	}
 	// Unassign the Egress IP from the local Node if it was assigned by the agent.
-	if _, err := c.ipAssigner.UnassignIP(eState.egressIP); err != nil {
+	egress, _ := c.egressLister.Get(egressName)
```

Comment:
it can pass the egress via a parameter of the func, instead of querying it another time.

Reply:
sure will do that.